File: ios/Classes/AppOpen/ContentCallback.swift
//
//  AppOpenMethodHandler.swift
//  clever_ads_solutions
//
//  Copyright © 2024 CleverAdsSolutions LTD, CAS.AI. All rights reserved.
//

import CleverAdsSolutions

class ContentCallback: MappedCallback, CASPaidCallback {
    func willShown(ad: any CASImpression) {
        invokeMethod("onShown", ad.toDict())
    }

    func didShowAdFailed(error: String) {
        invokeMethod("onShowFailed", ["error": error])
    }

    func didClickedAd() {
        invokeMethod("onClicked")
    }

    func didPayRevenue(for ad: any CASImpression) {
        invokeMethod("onAdRevenuePaid", ad.toDict())
    }

    func didClosedAd() {
        invokeMethod("onClosed")
    }
}

File: android/src/main/kotlin/com/cleveradssolutions/plugin/flutter/appopen/ContentCallback.kt
package com.cleveradssolutions.plugin.flutter.appopen

import com.cleveradssolutions.plugin.flutter.bridge.base.MappedCallback
import com.cleveradssolutions.plugin.flutter.bridge.base.MappedMethodHandler
import com.cleveradssolutions.plugin.flutter.util.toMap
import com.cleversolutions.ads.AdImpression
import com.cleversolutions.ads.AdPaidCallback

class ContentCallback(
    handler: MappedMethodHandler<*>,
    id: String
) : MappedCallback(handler, id), AdPaidCallback {

    override fun onShown(ad: AdImpression) {
        invokeMethod("onShown", ad.toMap())
    }

    override fun onShowFailed(message: String) {
        invokeMethod("onShowFailed", "error" to message)
    }

    override fun onClicked() {
        invokeMethod("onClicked")
    }

    override fun onAdRevenuePaid(ad: AdImpression) {
        invokeMethod("onAdRevenuePaid", ad.toMap())
    }

    override fun onClosed() {
        invokeMethod("onClosed")
    }

}